openrarity/cli/opensea.py

```diff
@@ -65,42 +85,43 @@ def fetch_assets(
             case _:
                 raise ValueError("Input file must be either a .txt or .json file.")
 
-    tokens = aio.run(
-        OpenseaApi.fetch_opensea_assets_data(slug=slug, token_ids=token_ids)
-    )
+    file_exists = False
+    overwrite_flag = False
+    rank_with_existing_assets_flag = False
+    if output:
+        tokens_path = output / f"{slug}_opensea.json"
+        # Check the existance of token collection file and populate `file_exists`,`overwrite_flag` and `rank_with_existing_assets_flag` accordingly.
+        file_exists,overwrite_flag,rank_with_existing_assets_flag = check_file_existence(tokens_path,rank)
+
+    # Fetch token collection data from opensea api or read from the tokens path.
+    if output and rank_with_existing_assets_flag:
+        tokens_path = output / f"{slug}_opensea.json"
+        contents = tokens_path.read_text()
+        tokens = json.loads(contents)
+    else:
+        tokens = aio.run(
+            OpenseaApi.fetch_opensea_assets_data(slug=slug, token_ids=token_ids)
+        )
+
     ranks = None
     if rank:
         ranks = TokenCollection(
             "non-fungible" if not semi_fungible else "semi-fungible", tokens
         ).rank_collection()
 
     column_values = columns.split(",")  # type: ignore
-    if write:
-        tokens_path = dir / f"{slug}_opensea.json"
-        tokens_path.write_text(json.dumps(tokens, indent=2))
-        typer.echo(f"Writing {str(tokens_path)}...")
+    if output is not None:
+        if overwrite_flag or not file_exists:
+            tokens_path = output / f"{slug}_opensea.json"
+            tokens_path.write_text(json.dumps(tokens, indent=2))
+            typer.echo(f"Writing {str(tokens_path)}...")
+
         if ranks is not None:
-            ranks = [[str(row[c]) for c in columns] for row in ranks]  # type: ignore
-            ranks_path = dir / f"{slug}_ranks.json"
+            ranks = [[str(row[c]) for c in column_values] for row in ranks]  # type: ignore
+            ranks_path = output / f"{slug}_ranks.json"
             ranks_path.write_text(json.dumps(ranks, indent=2))
             typer.echo(f"Writing {str(ranks_path)}...")
     elif ranks is not None:
         print_rankings(ranks, column_values)
     else:
         print(json.dumps(tokens))
```

Comment:
I think we can clean this up and be a little more succinct with the if else logic. Right now it is hard to follow what is happening.

Reply:
Try grouping the code a little differently. Have all the read/write/fetch for the token assets happen first then go through the ranking/writing logic.